models/models_methods.go

```diff
@@ -304,12 +304,23 @@ func (workflow *WorkflowModel) Validate(secrets ...[]envmanModels.EnvironmentIte
 			if err != nil {
 				return warnings, err
 			}
-			if s := opts.IsSensitive; (s != nil && *s) && (opts.IsExpand != nil && !*opts.IsExpand) {
+
+			isSensitive := opts.IsSensitive
+			if isSensitive == nil {
+				isSensitive = pointers.NewBoolPtr(envmanModels.DefaultIsSensitive)
+			}
+
+			isExpand := opts.IsExpand
+			if isExpand == nil {
+				isExpand = pointers.NewBoolPtr(envmanModels.DefaultIsExpand)
+			}
+
+			if *isSensitive && !*isExpand {
 				return warnings, fmt.Errorf("is_sensitive option set to true but is_expand is not, sensitive inputs cannot have direct values and to be able to use environment variable for input: (%s) you need to enable is_expand", key)
 			}
 
-			if s := opts.IsSensitive; len(secrets) >= 1 && (s != nil && *s) {
-				if err := isSecretEnv(value, secrets[0]); err != nil {
+			if *isSensitive && secrets != nil {
+				if err := isSecretEnv(value, secrets); err != nil {
```

Comment:
1. if secrets len == 0, we do not have any secrets - fail
2. if secrets len > 0 do your validation